### DISCUSSION TITLE
Old scheduler sometimes is still running after deploy

### DISCUSSION THREAD

**Discussion by Bykiev** [quartznet/quartznet#1452] [Q&A]
Hi, I found some strange issue: we're hosting our ASP.NET Web Api project on IIS. Sometimes after deploing the new version old scheduler is not stopping and 2 jobs are executing simultaneously. 

I'm using Quartz.NET with Autofac.Extras.Quartz library. I believe I'm doing smth wrong. Maybe I should call `Shutdown` method on `Application_End` event?

My configuration is looking this way:

```
public class JobScheduler
{
        public async Task Initialize()
        {
            if (!_scheduler.IsStarted)
                await _scheduler.Start();

          // schedule my jobs here
        }
}
```

Autofac configuration:

```
            var schedulerConfig = (NameValueCollection)ConfigurationManager.GetSection("quartz");
            builder.RegisterModule(new QuartzAutofacFactoryModule
            {
                ConfigurationProvider = c => schedulerConfig
            });

            builder.RegisterModule(new QuartzAutofacJobsModule(typeof(MyJob).Assembly));
            builder.RegisterType<JobScheduler>().AsSelf();

           var container = builder.Build();
           
             var scheduler = container.Resolve<JobScheduler>();
            await scheduler.Initialize();
```

**Comment by lahma** [comment]
I haven't used the Autofac integration, but `Application_End`  sounds like a correct place to make sure resources are being disposed properly. You should resolve scheduler there and call `Shutdown` with parameter whether to wait for jobs to complete (beware that IIS might kill the process is shutdown takes too long). I'm unsure if there's something built-in like this for Autofac integration.
